src/Branch.ts

```diff
@@ -504,16 +552,88 @@ export class Branch {
 			sort,
 			start
 		});*/
+
+		const filtersMustSets: string[][] = [];
+		const filtersMustNotSets: string[][] = [];
+		// const filtersShouldSets: string[][] = [];
+		let filtersMustIds: string[] = [];
+		let filtersMustNotIds: string[] = [];
+		// let filtersShouldIds: string[] = [];
+		if (
+			(Array.isArray(filters) && isFilter(filters))
+			|| isFilter(filters)
+		) {
+			const filtersArray = forceArray(filters);
+
+			for (let filtersIndex = 0; filtersIndex < filtersArray.length; filtersIndex++) {
+				const filter = filtersArray[filtersIndex];
+				if (isBooleanFilter(filter)) {
+					const must = (filter as BooleanFilter).boolean.must ? forceArray((filter as BooleanFilter).boolean.must): [];
+					const mustNot = (filter as BooleanFilter).boolean.mustNot ? forceArray((filter as BooleanFilter).boolean.mustNot): [];
+					for (let mustIndex = 0; mustIndex < must.length; mustIndex++) {
+						const mustFilter = must[mustIndex];
```

Comment:
```suggestion
					for (const mustFilter of must) {
```

Reply:
I guess since this is a test tool, and should prob never get test-data with million entries, I could have just used forEach and not worried about performance.

Reply:
performance vs readability